### PR TITLE
[Bugfix][NPU] Skip FRACTAL_NZ format for MoE weights with unaligned dimensions

### PR DESCRIPTION
**Pull request by adityavaid**
#### Issue : https://github.com/sgl-project/sglang/issues/21201
PR #15904 introduced early weight transpose + FRACTAL_NZ cast for NPU MoE performance. The Ascend \`npu_grouped_matmul\` kernel with NZ format requires:
- **BF16/FP16:** both K and N divisible by 16
- **INT8:** K divisible by 16, N divisible by 32
Models like GPT-OSS-120B have \`intermediate_size_per_partition=360\` (360 % 16 != 0), which causes a hard crash. This fix gracefully falls back to ND format for those weights while preserving the NZ optimization for models with aligned dimensions.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/layers/quantization/unquant.py [hunk 1]
```diff
@@ -317,9 +320,26 @@ def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
             for weight_name in ["w13_weight", "w2_weight"]:
                 weight = getattr(layer, weight_name)
                 weight.data = weight.data.transpose(1, 2)
-                weight.data = npu_format_cast(
-                    weight.data,
-                )
+                k, n = weight.data.shape[-2], weight.data.shape[-1]
```
Comment (OrangeRedeng) [inline python/sglang/srt/layers/quantization/unquant.py]: Hi! Thanks for your fix! But maybe we should move this logic to the ```npu_format_cast()``` function, because we use it not only in unquant.py, but in all quant_kernels cases (such as w8a8, w4a4 and etc.)

**Review by OrangeRedeng — APPROVED** [review]

File: python/sglang/srt/hardware_backend/npu/utils.py [hunk 2]
```diff
@@ -135,12 +157,24 @@ def npu_format_cast(
             "significantly reduced."
         )
         return tensor
-    else:
-        return torch.ops.npu.npu_format_cast(tensor, acl_format.value)
+
+    if acl_format == NPUACLFormat.ACL_FORMAT_FRACTAL_NZ and not _is_nz_aligned(tensor):
+        k, n = tensor.shape[-2], tensor.shape[-1]
+        logger.warning(
```
Comment (OrangeRedeng) [inline python/sglang/srt/hardware_backend/npu/utils.py]:
```suggestion
        logger.warning_once(
```
please use ```logger.warning_once(``` insted ```logger.warning(``` in order not to fill the console with garbage

**Review by OrangeRedeng — APPROVED** [review]

**Comment by adityavaid** [comment]
@ping1jing2 Is this happening because of the labels ? Either way something is wrong with CI pipeline

**Review by ping1jing2 — APPROVED** [review]
